src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContext.cs

```diff
@@ -49,6 +50,10 @@ public DotNetProjectContext(ProjectContext projectContext, string configuration,
             // Workaround https://github.com/dotnet/cli/issues/3164
             IsClassLibrary = !(_compilerOptions.EmitEntryPoint
                     ?? projectContext.ProjectFile.GetCompilerOptions(null, configuration).EmitEntryPoint.GetValueOrDefault());
+
+            var dependencyProvider = new DotNetDependencyProvider(_projectContext);
```

Comment:
How about adding a builder method `.WithDependencyProvider` ? If there is no DependencyProvider specified, we don't expose any dependencies at all.